remoteappmanager/db/csv_db.py

```diff
@@ -0,0 +1,149 @@
+""" This module provides support for using CSV file as the database
+of the remoteappmanager.
+
+Expect the first row of the CSV file to contain headers and
+the headers should contain the followings (not in order):
+
+user.name : str
+    Name of the user
+
+application.image : str
+    Image name of the application
+
+policy.allow_home : int
+    Is home directory mounted (1: true, 0: false)
+
+policy.allow_view : int
+    Is the application viewable by others (1: true, 0: false)
+
+policy.allow_common : int
+    Is the common data volume available (1: true, 0: false)
+    If true, requires definitions of volume_source, volume_target
+    and volume_mode
+
+policy.volume_source : str
+    Path for the common data volume on the host machine
+    If undefined, common data volume is not available
+
+policy.volume_target : str
+    Target mount point of the common data volue in the application
+    If undefined, common data volume is not available
+
+policy.volume_mode : str
+   Mode for read/write access (ro: read-only, rw: read-write)
+   If undefined, common data volume is not available
+
+.. note::
+
+- Additional columns are ignored.
+- This reader does not try to eliminate duplicated application and policy
+  record for a given user.  It faithfully reads the CSV file and returns
+  what it says.
+
+"""
+
+import csv
+
+from remoteappmanager.db.interfaces import (
+    ABCAccounting, ABCApplication, ABCApplicationPolicy)
+
+
+class CSVApplication(ABCApplication):
+    pass
+
+
+class CSVApplicationPolicy(ABCApplicationPolicy):
+    pass
+
+
+# FIXME: The user object could simply be a str (the username)
+# We need this because HomeHandler._start_container takes an object with
+# the username in its `name` attribute
+class CSVUser(object):
+
+    def __init__(self, name):
+        self.name = name
+
+
+_HEADERS = ('user.name',
+            'application.image',
+            'policy.allow_home',
+            'policy.allow_view',
+            'policy.allow_common',
+            'policy.volume_source',
+            'policy.volume_target',
+            'policy.volume_mode')
+
+
+class CSVAccounting(ABCAccounting):
+    """ Accounting class that reads a CSV file and is used by the
+    remoteappmanager.  Currently only accepts one csv file.
+    """
+
+    def __init__(self, csv_file_path, **kwargs):
+        """ Initialiser
+
+        Parameters
+        ----------
+        csv_file_path : str
+            File path for the CSV file
+
+        Optional parameters
+        -------------------
+        kwargs : dict
+            optional parameters for open(csv_file_path, **kwargs)
```

Comment:
Apparently you could: http://www.sphinx-doc.org/en/stable/ext/example_numpy.html
But I also remember running into similar problem related to `**`